src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostAbortedExceptionTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace Microsoft.Extensions.Hosting.Unit.Tests
+{
+    public sealed class HostAbortedExceptionTests
+    {
+        [Fact]
+        public void TestEmptyException()
+        {
+            var exception = new HostAbortedException();
+            Assert.Null(exception.InnerException);
+            Assert.Throws<HostAbortedException>(new Action(() => throw exception));
+        }
+
+        [Theory]
+        [InlineData("Host aborted.", false)]
+        [InlineData("Host aborted.", true)]
+        public void TestException(string? message, bool innerException)
+        {
+            HostAbortedException exception = innerException
+                ? new HostAbortedException(message, new Exception())
+                : new HostAbortedException(message);
+
+            Assert.Equal(message, exception.Message);
+            if (innerException)
+            {
+                Assert.NotNull(exception.InnerException);
+            } 
+            
+            HostAbortedException thrownException = Assert.Throws<HostAbortedException>(new Action(() => throw exception));
+            Assert.Equal(message, thrownException.Message);
+        }
+
+        [Fact]
+        public void TestSerialization()
+        {
+            var exception = new HostAbortedException();
+            using var serializationStream = new MemoryStream();
+
+            var formatter = new BinaryFormatter();
```

Comment:
I believe we aren't adding any more binary serialization even to exceptions. @GrabYourPitchforks  what's our policy?

Reply:
See https://github.com/dotnet/runtime/issues/60600#issuecomment-1128095213
I am not sure if we need this.

Reply:
I'm always going to support efforts to avoid slapping `[Serializable]` on more types if the product team thinks that serializing them is not a legitimate scenario. :)

Reply:
Generally for exceptions we (at least used to) make them serializable as a matter of course, the canonical scenario being something like, your app or something it called threw an exception of any type (at least a type in core libraries code) and you could serialize it and pass it to some diagnostic and point. That was the scenario that originally motivated us to bring back binary serialization for all our exceptions (in 2.0?)
That scenario stands or falls on “exceptions generally binary serialized with fidelity”. So the question is: do we care about that scenario or not: if not, we should drive to remove binary serialization from all our exceptions, at least unless there’s some other compelling scenario. That’s why I brought it up with you here @GrabYourPitchforks.
My assumption is that we are removing binary serialization from all our exceptions either this release or next. I’m guessing it’s next at this point: I don’t have the plan in front of me.
Given it’s on the way out I’d suggest to not make any more exceptions binary serializable if we can. We are moving off this tech.

Reply:
@danmoseley In https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#no-new-serializable-types-introduced-net-6

It starts off agreeing with you:

> Beginning with .NET 6.0, no new types will be marked [Serializable], nor will the [Serializable] attribute be added to existing types or to types ported from .NET Framework.

then moves into disagreement:

> The exclusion is for dotnet org first-party projects which target netstandard2.0 or earlier. For these projects, Exception-derived types should continue to be annotated as [Serializable] so that they may be remoted across app domain boundaries.

Assuming that this gets added to a `netstandard2.0` ref, then the netfx lib (or netstandard2.0 lib if it's not split-built) should still, per that doc, be declaring the exception as `[Serializable]` in case netfx is using it with something like cross-AppDomain catches/logging.

Unless we want to change the plan, or make an exclusion-exclusion, of course.

Reply:
Thanks for the judicial analysis 🙂 makes sense. In due course then, our binary formatter test will be pared down not to zero, but to only types exposed in a NS2.0 ref, and the test will then only run on .NET Framework.

Reply:
At the latest until Windows decides to EOL Net FX we can remove it 😄